provider/search-provider/src/main/java/dev/ikm/tinkar/provider/search/Searcher.java

```diff
@@ -366,4 +375,33 @@ public static List<PublicId> membersOf(PublicId memberPatternId) {
         }
         return Collections.emptyList();
     }
+
+    /**
+     * Returns PublicId for the Concepts associated with the identifier
+     *
+     * @param   identifier UUID of the Semantic
+     * @return  PublicId for the Concept associated with the identifier
+     */
+
+    public static Optional<PublicId> getPublicId(String identifier) {
```

Comment:
@dettmoney, we expect the String parameter to be an SDO identifier, such as a Snomed ID, rather than a UUID String. This difference will have downstream effects on how the PublicId lookup is performed, so there will need to be some refactoring here to adjust to the parameter expectations.

Reply:
I think we have a still-open discussion topic about how to improve search. One possibility is having a specific Lucent index for SDO identifiers... I'd recommend we not use the search function as it currently exists, but to iterate through the identifier pattern semantics for a match.

Reply:
As a comment on @rbradley813 's comment about the String parameter expected to be a Snomed ID -- can we strongly type this parameter so there's no ambiguity (e.g. a SnomedId type that just wraps a String)?

Reply:
I pushed a new commit that addresses concerns from the code review comments. Please have another look at your leisure.

Reply:
The getPublicId(String identifier) should be two parameters not one to support searching for SNOMED IDs, LOINC IDs, and similar. If you make the first parameter the public id of the Source or Authority (SNOMED, LOINC, RxNorm, HL7), then that can iterate over the identifier pattern and have an expectation of a single match.  Some identifiers schemes use mnemonics, and they may be overlap across schemes, and so this API call is broken, and does not reflect the domain of identifiers. 

@rbradley813 Please ensure this call is updated to be consistent with the expected behavior of the identifier pattern.

Reply:
Optional<PublicId> getPublicId(PublicId identifierSource, String identifierValue). 

And the TikarTerm should have EntityProxy.Concept values for the identifier schemes that can be passed in... 

For example, SNOMED CT integer ID: [87360947-e603-3397-804b-efd0fcc509b9] should be in TinkarTerm, and easily referencable.